corehq/apps/sso/utils/entra.py

```diff
@@ -0,0 +1,48 @@
+import requests
+
+from corehq.apps.sso.exceptions import EntraVerificationFailed
+
+
+class MSGraphIssue:
+    HTTP_ERROR = "http_error"
+    VERIFICATION_ERROR = "verification_error"
+    EMPTY_ERROR = "empty_error"
+
+
+def get_all_members_of_the_idp_from_entra(idp):
+    import msal
+    authority_base_url = "https://login.microsoftonline.com/"
+    authority = f"{authority_base_url}{idp.api_host}"
+
+    config = {
+        "authority": authority,
+        "client_id": idp.api_id,
+        "scope": ["https://graph.microsoft.com/.default"],
+        "secret": idp.api_secret,
+        "endpoint": "https://graph.microsoft.com/v1.0/users"
+    }
+
+    # Create a preferably long-lived app instance which maintains a token cache.
+    app = msal.ConfidentialClientApplication(
+        config["client_id"], authority=config["authority"],
+        client_credential=config["secret"],
+    )
+    # looks up a token from cache
+    result = app.acquire_token_silent(config["scope"], account=None)
+    # try:
```

Comment:
remove this comment?

Reply:
Thank you for catching this! Removed in 674be9f4176fe1f39c8b04348be97cd4974203f4